tests/main/cgroup-devices-v2/task.yaml

```diff
@@ -0,0 +1,119 @@
+  - -centos-7-*
+  - -centos-8-*
+
+prepare: |
+    tests.session -u test prepare
+
+restore: |
+    tests.session -u test restore
+
+execute: |
+    dump_cgroup_device_progs() {
+        # dump all progs but those that are assigned to systemd, in which case
+        # they have the pids list non empty
+        bpftool prog list -j | jq -r '.[] | select(.type == "cgroup_device") | select(.pids == null) | .id'
+    }
+    echo "Dump BPF programs that are of type cgroup_device"
+    dump_cgroup_device_progs > cgroup_device.progs-1
+
+    echo "Install a test service"
+    # Because this service is of type "simple" it is considered "ready" instantly.
+    # In reality the process needs to go through snap "run" chain to be really
+    # ready. As a workaround, touch a "remove-me" file that is removed by the
+    # service on startup, restart the service and the wait for the file to
+    # disappear.
+    mkdir -p /var/snap/test-snapd-service/common
+    touch /var/snap/test-snapd-service/common/remove-me
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-service
+    # Wait for the service to really be alive and running. Otherwise the "main pid"
+    # will be still tracking snap-run-confine-exec chain and be unreliable.
+    retry -n 5 test ! -e /var/snap/test-snapd-service/common/remove-me
+
+    echo "Extract the PID of the main process tracked by systemd"
+    # It would be nicer to use "systemctl show --property=... --value" but it doesn't work on older systemd.
+    pid=$(systemctl show snap.test-snapd-service.test-snapd-service.service --property=ExecMainPID | cut -d = -f 2)
+
+    echo "Extract the cgroup of the main process"
+    cgroup=$(grep 0:: < "/proc/$pid/cgroup" | cut -d : -f 3)
+
+    # The service always runs in a device cgroup created by systemd (or in a
+    # scope created snap run for user processes)
+    test "$cgroup" = /system.slice/snap.test-snapd-service.test-snapd-service.service
+
+    # Initially since there are no devices assigned to snap, there is no map with device access
+    test ! -e /sys/fs/bpf/snap/snap_test-snapd-service_test-snapd-service
+    # Do another dump
+    dump_cgroup_device_progs > cgroup_device.progs-2
+    echo "Verify no new device cgroup programs are present"
+    test "$(wc -l < cgroup_device.progs-1)" = "$(wc -l < cgroup_device.progs-2)"
+
+    echo "Connect the joystick interface"
+    snap connect test-snapd-service:joystick
+
+    # a device cgroup is not added until the service gets restarted
+    test ! -e /sys/fs/bpf/snap/snap_test-snapd-service_test-snapd-service
+
+    echo "Run /bin/true via snap-confine, so that we create the device cgroup made by snapd"
+    snap run test-snapd-service.sh -c /bin/true
+    # with new devices assigned to the service, we see a map listing the devices
+    # that access is allowed to
+    echo "Verify that device access map for the application is present and is pinned to a file"
+    test -e /sys/fs/bpf/snap/snap_test-snapd-service_sh
```

Comment:
Should we have a 

     test ! -e /sys/fs/bpf/snap/snap_test-snapd-service_sh

before running `/bin/true`, if we expect this file to appear only after running the snap?